tests/test_cache.py

```diff
@@ -354,30 +275,35 @@ def __call__(self, **kwargs):
     cache = AgentCache()
     value = "test too"
     resource = Id("test::Resource", "test", "key", "test", 100).get_instance()
-    version = 100
-    cache.open_version(version)
-    assert None is cache.get_or_else("test", creator, resource=resource, version=version, cache_none=False, param=None)
+
+    # Check for 2 successive cache miss since caching None is disabled
+    assert None is cache.get_or_else("test", creator, resource=resource, cache_none=False, param=None)
     assert len(called) == 1
-    assert None is cache.get_or_else("test", creator, resource=resource, version=version, cache_none=False, param=None)
+    assert None is cache.get_or_else("test", creator, resource=resource, cache_none=False, param=None)
     assert len(called) == 2
-    assert value == cache.get_or_else("test", creator, resource=resource, version=version, cache_none=False, param=value)
-    assert value == cache.get_or_else("test", creator, resource=resource, version=version, cache_none=False, param=value)
+
+    # Check for 1 cache miss and then a hit.
+    assert value == cache.get_or_else("test", creator, resource=resource, cache_none=False, param=value)
+    assert value == cache.get_or_else("test", creator, resource=resource, cache_none=False, param=value)
     assert len(called) == 3
 
+    # The Sequencer will return the next item in the sequence on each cache miss
     seq = Sequencer([None, None, "A"])
-    assert None is cache.get_or_else("testx", seq, resource=resource, version=version, cache_none=False)
+
+    # Check that we have cache misses until the sequencer returns a non-None value.
+    assert None is cache.get_or_else("testx", seq, resource=resource, cache_none=False)
     assert seq.count == 1
-    assert None is cache.get_or_else("testx", seq, resource=resource, version=version, cache_none=False)
+    assert None is cache.get_or_else("testx", seq, resource=resource, cache_none=False)
     assert seq.count == 2
-    assert "A" == cache.get_or_else("testx", seq, resource=resource, version=version, cache_none=False)
+    assert "A" == cache.get_or_else("testx", seq, resource=resource, cache_none=False)
     assert seq.count == 3
-    assert "A" == cache.get_or_else("testx", seq, resource=resource, version=version, cache_none=False)
+    assert "A" == cache.get_or_else("testx", seq, resource=resource, cache_none=False)
     assert seq.count == 3
-    assert "A" == cache.get_or_else("testx", seq, resource=resource, version=version, cache_none=False)
+    assert "A" == cache.get_or_else("testx", seq, resource=resource, cache_none=False)
     assert seq.count == 3
 
 
-def test_decorator():
+async def test_decorator(agent_cache):
```

Comment:
I don't think we need the `agent_cache` fixture here. We can just call into `AgentCache()`. The latter will be faster as it doesn't start a server and an agent.